modules/rostests/rosautotest/main.cpp

```diff
@@ -87,6 +89,12 @@ extern "C" int
 wmain(int argc, wchar_t* argv[])
 {
     int ReturnValue = 1;
+    DWORD TestStartTime, TestEndTime;
+
+    GetModuleFileNameW(NULL, TestName, _countof(TestName));
+    WCHAR* Name = wcsrchr(TestName, '\\');
+    Name++;
+    StringCchCopyW(TestName, _countof(TestName), Name);
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/strsafe/nf-strsafe-stringcchcopyw#remarks
> Behavior is undefined if the strings pointed to by pszSrc and pszDest overlap.

So for example, `GetModuleFileNameW` returns `C:\rosautotest.exe\0` in `TestName`, `Name` points to the "r" (i.e. `&TestName[3]`), and you then copy what's in the buffer starting at position 3 down to position 0 (so, overlapping copy).

Reply:
I removed the "StringCchCopyW" function and replaced it with a "memmove" which handles overlapping copies.
This allowed me to remove the include for the "strsafe.h" since this was the only strsafe function used. Thanks.